CondTools/SiStrip/plugins/SiStripBadChannelPatcher.cc

```diff
@@ -0,0 +1,228 @@
+  edm::LogInfo("SiStripBadChannelPatcher") << "... creating dummy SiStripBadStrip Data";
+
+  // this is the output object
+  auto obj = std::make_unique<SiStripBadStrip>();
+
+  // get the cabling object
+  detCabling_ = &iSetup.getData(cablingToken_);
+
+  // get the strips to add
+  std::vector<uint32_t> detids;
+  const auto& payload = iSetup.getData(badStripToken_);
+  payload.getDetIds(detids);
+
+  // copy the exisiting channels (excluding the ones to remove)
+  for (const auto& id : detids) {
+    // check on the detids to exclude
+    if (std::count(detIdsToExclude_.begin(), detIdsToExclude_.end(), id)) {
+      edm::LogInfo("SiStripBadChannelPatcher") << "I AM GOING TO EXCLUDE DETID: " << id;
+      continue;
+    } else {
+      LogDebug("SiStripBadChannelPatcher") << "I AM GOING TO KEEP DETID: " << id;
+    }
+
+    // check on the FEDs to exclude
+    const auto& currentFED = this->fedFromDetId(id);
+    if (std::count(fedsToExclude_.begin(), fedsToExclude_.end(), currentFED)) {
+      edm::LogInfo("SiStripBadChannelPatcher") << "I AM GOING TO EXCLUDE DETID: " << id;
+      continue;
+    } else {
+      LogDebug("SiStripBadChannelPatcher") << "I AM GOING TO KEEP DETID: " << id;
+    }
+
+    SiStripBadStrip::Range range = payload.getRange(id);
+    std::vector<unsigned int> theSiStripVector;
+    unsigned int theBadStripRange;
+    for (std::vector<unsigned int>::const_iterator badStrip = range.first; badStrip != range.second; ++badStrip) {
+      unsigned short firstBadStrip = payload.decode(*badStrip).firstStrip;
+      unsigned short NconsecutiveBadStrips = payload.decode(*badStrip).range;
+      theBadStripRange = obj->encode(firstBadStrip, NconsecutiveBadStrips);
+      theSiStripVector.push_back(theBadStripRange);
+    }
+    SiStripBadStrip::Range outRange(theSiStripVector.begin(), theSiStripVector.end());
+    if (!obj->put(id, outRange))
+      edm::LogError("SiStripBadChannelPatcher") << "[SiStripBadChannelPatcher::analyze] detid already exists";
+  }  // loop on the detids of the original payload
+
+  const auto& detInfo =
+      SiStripDetInfoFileReader::read(edm::FileInPath{SiStripDetInfoFileReader::kDefaultFile}.fullPath());
+
+  // add to the list of DetIds to include also the one from the list of FEDs
+  const std::vector<uint32_t>& allDetIds = detInfo.getAllDetIds();
+  this->addDetIdsFromExcludedFEDs(allDetIds);
+
+  // add more full bad detids
+  if (!detIdsToInclude_.empty()) {
+    edm::LogInfo("SiStripBadChannelPatcher") << "I AM GOING TO ADD MORE DETIDS";
+
+    std::stringstream ss;
+    for (const auto& detid : detIdsToInclude_) {
+      edm::LogInfo("SiStripBadChannelPatcher") << "I AM GOING TO ADD DETID: " << detid;
+      const auto& nAPVs = detInfo.getNumberOfApvsAndStripLength(detid).first;
+
+      std::vector<unsigned int> theSiStripVector;
+      unsigned short firstBadStrip{0}, NconsecutiveBadStrips{0};
```

Comment:
If one wants to adhere to the prescriptions for the local variable names...
```suggestion
      unsigned short firstBadStrip{0}, nConsecutiveBadStrips{0};
```